benches/bin/mvcctxn.rs

```diff
@@ -0,0 +1,248 @@
+    config.lockcf.write_buffer_size = ReadableSize::gb(1);
+
+    let cf_ops = config.build_cf_opts();
+
+    let db = Arc::new(new_engine_opt(path, DBOptions::new(), cf_ops).unwrap());
+
+    for _ in 0..versions {
+        for key in keys {
+            let value = vec![0u8; value_len];
+            let start_ts = next_ts();
+            let commit_ts = next_ts();
+
+            prewrite(
+                Arc::clone(&db),
+                &[Mutation::Put((Key::from_raw(key), value))],
+                key,
+                start_ts,
+            );
+            commit(Arc::clone(&db), &[Key::from_raw(key)], start_ts, commit_ts);
+        }
+    }
+    db
+}
+
+#[inline]
+fn get(db: Arc<DB>, key: &Key, statistics: &mut Statistics) -> Option<Vec<u8>> {
+    let snapshot = get_snapshot(Arc::clone(&db));
+    let start_ts = next_ts();
+    let snapstore = SnapshotStore::new(snapshot, start_ts, IsolationLevel::SI, true);
+    snapstore.get(key, statistics).unwrap()
+}
+
+fn bench_get(db: Arc<DB>, keys: &[Vec<u8>]) -> f64 {
+    let mut fake_statistics = Statistics::default();
+    let mut rng = rand::thread_rng();
+    do_bench(
+        || {
+            let key = rng.choose(keys).unwrap();
+            let key = Key::from_raw(key);
+
+            get(Arc::clone(&db), &key, &mut fake_statistics).unwrap()
+        },
+        500000,
+    )
+}
+
+fn bench_set(db: Arc<DB>, keys: &[Vec<u8>], value_len: usize) -> f64 {
+    let mut rng = rand::thread_rng();
+    do_bench(
+        || {
+            let start_ts = next_ts();
+            let commit_ts = next_ts();
+            let value = vec![0u8; value_len];
+
+            let key = rng.choose(keys).unwrap();
+
+            prewrite(
+                Arc::clone(&db),
+                &[Mutation::Put((Key::from_raw(key), value))],
+                key,
+                start_ts,
+            );
+            commit(Arc::clone(&db), &[Key::from_raw(key)], start_ts, commit_ts)
+        },
+        500000,
+    )
+}
+
+fn bench_delete(db: Arc<DB>, keys: &[Vec<u8>]) -> f64 {
+    let mut rng = rand::thread_rng();
+    do_bench(
+        || {
+            let start_ts = next_ts();
+            let commit_ts = next_ts();
+
+            let key = rng.choose(keys).unwrap();
+            prewrite(
+                Arc::clone(&db),
+                &[Mutation::Delete(Key::from_raw(key))],
+                key,
+                start_ts,
+            );
+            commit(Arc::clone(&db), &[Key::from_raw(key)], start_ts, commit_ts)
+        },
+        500000,
+    )
+}
+
+// Run all bench with specified parameters
+fn bench_single_row(
+    table_size: usize,
+    version_count: usize,
+    data_len: usize,
+    bench_type: &BenchType,
+) {
+    let (mut keys, value_len, log_name) = match *bench_type {
+        BenchType::Row => (generate_row_keys(1, 0, table_size), data_len, "row"),
+        BenchType::UniqueIndex => (
+            generate_unique_index_keys(1, 1, data_len, table_size),
+            8,
```

Comment:
Why 8 here?

Reply:
In a real unique index, the value of the key-value pair stores RowID which is an int64